plugins/reminders.py

```diff
@@ -0,0 +1,164 @@
+    time_arg = args.next_arg()
+    if not isinstance(time_arg, plugins.commands.StringArg): return None
+    time_str = time_arg.text
+    seconds = 0
+    pos = 0
+    while (time_match := time_re.match(time_str, pos)) is not None:
+        pos = time_match.end()
+        seconds += int(time_match[1]) * time_expansion[time_match.lastgroup] # type: ignore
+    if not (pos == len(time_str) and seconds > 0): return None
+    return seconds
+
+def format_msg(guild_id: str, channel_id: str, msg_id: str) -> str:
+    return "https://discord.com/channels/{}/{}/{}".format(guild_id, channel_id, msg_id)
+
+def format_reminder(reminder: Reminder) -> str:
+    guild, channel, msg, send_time, contents = itemgetter("guild", "channel", "msg", "time", "contents")(reminder)
+    if contents == "": return "{} for <t:{}:F>".format(format_msg(guild, channel, msg), send_time)
+    return util.discord.format("{!i} ({}) for <t:{}:F>", contents, format_msg(guild, channel, msg), send_time)
+
+def format_text_reminder(reminder: Reminder) -> str:
+    guild, channel, msg, send_time, contents = itemgetter("guild", "channel", "msg", "time", "contents")(reminder)
+    return util.discord.format("{!i} ({}) for {}", contents, format_msg(guild, channel, msg),
+        time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime(int(send_time))))
+
+async def send_reminder(user_id: str, reminder: Reminder) -> None:
+    channel = discord_client.client.get_channel(int(reminder["channel"]))
+    if not isinstance(channel, discord.TextChannel):
+        logger.info("Could not send reminder {} for user {}".format(str(reminder), user_id))
+        return
+    try:
+        creation_time = discord.utils.snowflake_time(int(reminder["msg"])).replace(tzinfo=timezone.utc)
+        user = int(user_id)
+        await channel.send(util.discord.format(
+            "{!m} asked to be reminded <t:{}:R>: {}", user, int(creation_time.timestamp()), reminder["contents"]),
+            reference = discord.MessageReference(message_id = int(reminder["msg"]),
+                channel_id = int(reminder["channel"]), fail_if_not_exists = False),
+            allowed_mentions=discord.AllowedMentions(everyone = False, users = [discord.Object(user)], roles = False))
+    except discord.Forbidden:
+        logger.info("Could not send reminder {} for user {}".format(str(reminder), user_id))
+
+async def handle_reminders(user_id: str) -> None:
+    ref_time = time.time()
+    reminders = conf[user_id]
+    if reminders is None: return
+    await asyncio.gather(*(send_reminder(user_id, reminder)
+        for reminder in takewhile(lambda reminder: int(reminder["time"]) < ref_time, reminders)))
+    conf[user_id] = FrozenList(dropwhile(lambda reminder: int(reminder["time"]) < ref_time, reminders))
+
+async def run_reminder_interval() -> None:
+    for run_time in count(time.time(), 30):
+        await asyncio.sleep(run_time - time.time())
+        for user_id in conf: asyncio.create_task(handle_reminders(user_id))
```

Comment:
Why not do the thing we do in tickets:
https://github.com/discord-math/bot/blob/main/plugins/tickets.py#L1075-L1116